core/storage/leveldb/leveldb.cpp

```diff
@@ -77,21 +77,32 @@ namespace kagome::storage {
     std::string value;
     auto status = db_->Get(ro_, make_slice(key), &value);
     if (status.ok()) {
-      // FIXME: is it possible to avoid copying string -> Buffer?
+      // cannot move string content to a buffer
       return Buffer{}.put(value);
     }
 
-    // not always an actual error so don't log it
+    return status_as_error(status, logger_);
+  }
+
+  outcome::result<boost::optional<Buffer>> LevelDB::tryGet(
+      const Buffer &key) const {
+    std::string value;
+    auto status = db_->Get(ro_, make_slice(key), &value);
+    if (status.ok()) {
+      // cannot move string content to a buffer
+      return boost::make_optional(Buffer{}.put(value));
+    }
+
     if (status.IsNotFound()) {
-      return error_as_result<Buffer>(status);
+      return boost::none;
     }
 
-    return error_as_result<Buffer>(status, logger_);
+    return status_as_error(status, logger_);
   }
 
   bool LevelDB::contains(const Buffer &key) const {
     // here we interpret all kinds of errors as "not found".
-    // is there a better way?
+    // TODO(Harrm): for review: might it be better to throw?
```

Comment:
I don't think so. You may see there is a countless number of getting non existent keys at the syncing.

Reply:
What is meant here is to throw when encountered an error other than NO_VALUE.

Reply:
Because the way it is now, any error in the storage would make this method just return false.